packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/base/common_services/push_notifications_service.dart

```diff
@@ -0,0 +1,71 @@
+{{> licence.dart }}
+
+import 'package:firebase_messaging/firebase_messaging.dart';
+
+import '../../assets.dart';
+import '../models/errors/error_model.dart';
+import '../repositories/push_notification_repository.dart';
+
+class PushNotificationsService {
+  final PushNotificationRepository _pushNotificationRepository;
+  final FirebaseMessaging firebaseMessaging;
+
+  PushNotificationsService(
+    this._pushNotificationRepository,
+    this.firebaseMessaging,
+  );
+
+  Future<void> unsubscribe([bool setNotifications = false]) async {
+    final token = await _pushNotificationRepository.getToken();
+    if (token == null) {
+      throw ErrorModel();
+    }
+    await _pushNotificationRepository.unsubscribe(token);
+    await _pushNotificationRepository
+        .setNotificationSubscribed(setNotifications);
+  }
+
+  Future<bool> areNotificationsEnabled() async {
+    final permissionStatus = await firebaseMessaging.getNotificationSettings();
```

Comment:
Since all the interactions with firebase messaging happen through the `_pushNotificationRepository` maybe we can add the `getNotificationSettings()` method there to be consistent?